test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,224 @@
+        $specificationFileName       = 'specification.json';
+        $specificationArray          = $this->getSpecificationArray($specificationFileName);
+        $expectedSpecificationConfig = new SpecificationConfig(
+            $specificationArray['path'],
+            $specificationArray['type'] ?? null,
+            $specificationArray['name_space'],
+            $specificationArray['media_type']
+        );
+
+        $expectedSpecifications = [self::SPECIFICATION_NAME => $expectedSpecificationConfig];
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specifications = $this->specificationLoader->list();
+
+        Assert::assertEquals($expectedSpecifications, $specifications);
+
+        $specificationConfig = $this->specificationLoader->get(self::SPECIFICATION_NAME);
+        Assert::assertEquals($expectedSpecificationConfig, $specificationConfig);
+    }
+
+    public function testLoadYamlFromCache(): void
+    {
+        $specificationFileName = 'specification.yaml';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specification = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+
+        rename(__DIR__ . '/' . $specificationFileName, __DIR__ . '/' . $specificationFileName . '_test');
+        $specificationFromCache = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+        Assert::assertSame($specification, $specificationFromCache);
+        rename(__DIR__ . '/' . $specificationFileName . '_test', __DIR__ . '/' . $specificationFileName);
+    }
+
+    public function testLoadJsonFromCache(): void
+    {
+        $specificationFileName = 'specification.json';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specification = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+
+        rename(__DIR__ . '/' . $specificationFileName, __DIR__ . '/' . $specificationFileName . '_test');
+        $specificationFromCache = $this->specificationLoader->load(self::SPECIFICATION_NAME);
+        Assert::assertSame($specification, $specificationFromCache);
+        rename(__DIR__ . '/' . $specificationFileName . '_test', __DIR__ . '/' . $specificationFileName);
+    }
+
+    public function testLoadMissedSpecificationFileThrowsException(): void
+    {
+        $specificationFileName = 'specification.spec';
+        $specificationArray    = $this->getSpecificationArray($specificationFileName);
+
+        $this->specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $this->expectException(Throwable::class);
+        $this->expectExceptionMessage(sprintf('The file "%s" does not exist.', __DIR__ . '/' . $specificationFileName));
```

Comment:
The tested class depends on a FileLocatorInterface, and not on the concrete FileLocator.
Thus other implementations of the interface may never throw such exceptions, and the test will brake.
We should not test dependencies that are out of our control.